lms/djangoapps/learner_recommendations/tests/test_views.py

```diff
@@ -152,6 +154,162 @@ def test_successful_response(
         assert segment_mock.call_args[0][1] == "edx.bi.user.recommendations.viewed"
 
 
+class TestCrossProductRecommendationsView(APITestCase):
+    """Unit tests for the Cross Product Recommendations View"""
+
+    def setUp(self):
+        super().setUp()
+        self.associated_course_keys = ["ColumbiaX+BC24FNTC", "MITx+BLN"]
+
+    def _get_url(self, course_key):
+        """
+        Returns the url with a sepcific course id
+        """
+
+        return reverse_lazy(
+            "learner_recommendations:cross_product_recommendations",
+            kwargs={'course_id': f'course-v1:{course_key}+Test_Course'}
+        )
+
+    def _get_recommended_courses(self, has_location_restriction=False, restriction_num=0):
+        """
+        Returns an array of 2 discovery courses with or without country restrictions
+        """
+        courses = []
+        restriction_obj = {
+            "restriction_type": "blocklist",
+            "countries": ["CN"],
+            "states": []
+        }
+
+        for i in enumerate(self.associated_course_keys):
+            location_restriction = restriction_obj if has_location_restriction and restriction_num > 0 else None
+
+            courses.append({
+                "key": i[1],
+                "uuid": "6f8cb2c9-589b-4d1e-88c1-b01a02db3a9c",
+                "title": f"Title {i[0]}",
+                "image": {
+                        "src": "https://www.logo_image_url.com",
+                },
+                "url_slug": "https://www.marketing_url.com",
+                "course_type": "executive-education",
+                "owners": [
+                    {
+                            "key": "org-1",
+                            "name": "org 1",
+                            "logo_image_url": "https://discovery.com/organization/logos/org-1.png",
+                    },
+                ],
+                "course_runs": [
+                    {
+                        "key": "course-v1:Test+2023_T2",
+                        "marketing_url": "https://www.marketing_url.com",
+                        "availability": "Current",
+                    }
+                ],
+                "location_restriction": location_restriction
+            })
+
+            if restriction_num > 0:
+                restriction_num -= 1
```

Comment:
I think you are trying to mock one course with restriction and second without it. Instead of this can we just add the location restriction to one of the course?
```suggestion
            if has_location_restriction:
                courses[0]["location_restriction"] = location_restriction
```

Reply:
Yes you're right, but I also wanted to set up the `_get_recommended_courses` method so that depending on the number passed into the function, you can say how many courses you want to have a location restriction, and I also set-up a test case for when we have one course with a location restriction and then both courses with a country location, and that utilises the `restriction_num` argument passed in to dictate how many courses will get that location restriction object.